File: Container-With-Most-Water.cpp
class Solution {
public:
    int maxArea(vector<int>& v) {
        int l  = 0 , r = v.size() - 1;
        long long ans = 0 , len = v.size() -1; 
        while(l < r){
            int mn = min(v[l],v[r]);
            ans = max(ans,mn*len);

            if(v[l] <= v[r]) l++;    
             
             
            else if (v[l] > v[r])r--;
             

            len--;
        } 
        return ans;
    }
};

File: Container-With-Most-Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
const maxArea = (v) => {
    let l = 0, r = v.length - 1;
    let ans = 0;
    let width = v.length - 1;
    while (l < r) {
        const height = Math.min(v[l], v[r]);
        ans = Math.max(ans, height * width);
        
        v[l] <= v[r] ? l++ : r--;
        width--;
    }
    return ans;
};